### PR TITLE
Replace apps's colors with colors from the new guideline

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T299793

### Notes
* This PR sets the project to use the colors set in the Components package as part of our new components work
* @olga-ti we'll spin up a build for you to test, so you can check how the color substitutions are playing out. Feel free to comment here or in the phab task

### Test Steps
1.  Run the app, and check every possible view :)

### CODE REVIEW

File: Wikipedia.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -23,8 +23,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/wikimedia/wikipedia-ios-components.git",
       "state" : {
-        "branch" : "main",
-        "revision" : "81a362e601de6580c5dcbfa90bf128232b5d9d9f"
+        "branch" : "T299793-colors",
+        "revision" : "651799fc8ede64542c98eb90d72de67f12d428d2"
```
Comment (mazevedofs) [inline Wikipedia.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:27]: This was done so the build won't break, it should be reverted after the Components PR is merged https://github.com/wikimedia/wikipedia-ios-components/pull/4

File: Wikipedia/Third Party Code/SWStepSlider/SWStepSlider.swift [hunk 2]
```diff
@@ -24,16 +24,16 @@ open class SWStepSlider: UIControl {
```
Comment (tonisevener) [inline Wikipedia/Third Party Code/SWStepSlider/SWStepSlider.swift:24]: Slider colors may look different.

File: WMF Framework/Theme.swift [hunk 3]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+            return .gray800
+        case .black:
+            return .gray650
+        }
+    }
+
+    @objc public var hintWarningBackground: UIColor {
+        switch identifier {
+        case .light:
+            return .yellow600
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:243]: This should be `.yellow100`, according to spreadsheet.

File: WMF Framework/Theme.swift [hunk 4]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+        default:
+            return .red600
+        }
+    }
+    @objc public var warning: UIColor {
+        switch identifier {
+        case .light, .sepia, .widgetLight:
+            return .orange600
+        case .dark, .black, .widgetDark:
+            return .blue300
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:367]: This should be `.yellow600`, according to spreadsheet.

File: WMF Framework/Theme.swift [hunk 5]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+        default:
+            return .blue300
+
+        }
+    }
+
+    @objc public var tagSelectedBackground: UIColor {
+        switch identifier {
+        case .light:
+            return .blue600.withAlphaComponent(0.125)
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:584]: Should be 0.25 I think.

File: WMF Framework/Theme.swift [hunk 6]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+            return .blue300
+        }
+    }
+
+    @objc public var descriptionWarning: UIColor {
+        switch identifier {
+        case .light, .sepia, .widgetLight:
+            return .orange600
+        case .dark, .black, .widgetDark:
+            return .blue300
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:618]: Should be `.yellow600`.

File: WMF Framework/Theme.swift [hunk 7]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
-        self.diffContextItemBorder = diffContextItemBorder
-        self.diffMoveParagraphBackground = diffMoveParagraphBackground
-        self.diffCompareAccent = diffCompareAccent
-        self.diffCompareChangeHeading = diffCompareChangeHeading
-        self.talkPageBackground  = talkPageBackground
-        self.talkPageCoffeRollBackground = talkPageCoffeRollBackground
+    public var diffTextAdd: UIColor {
+        switch identifier {
+        case .light:
+            return .green100
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:676]: Seems like these are off from the spreadsheet, should be `.gray700` for light, and `green600` for the others.

**Review by tonisevener — COMMENTED** [review]
@mazevedofs Thanks for working through this!

Leaving comments of things that jumped out at me, just from a pure code review. I haven't looked at the old-style widget colors yet. Many of these are optional I would say, since they were one-off colors from before (like Settings icons). But I wanted to point them out as an area @olga-ti may want to look at closely, in case the difference is worth spinning up a new official color for.

File: WMF Framework/Theme.swift (hunk 3 above)
Comment (mazevedofs) [reply WMF Framework/Theme.swift:243]: The figma ref doesn't have a yellow100, I'll ask Olga about it

File: WMF Framework/Theme.swift [hunk 8]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+        case .light, .sepia, .widgetLight:
+            return .gray700
+        case .dark, .black, .widgetDark:
+            return .gray100
+        }
+    }
+
+    @objc public var secondaryText: UIColor {
+        switch identifier {
+        case .light:
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:277]: I would go ahead and add `.widgetLight` to this line as well, looks like `WMFTodayContinueReadingWidgetViewController` still uses it.

File: WMF Framework/Theme.swift [hunk 9]
```diff
@@ -169,580 +93,714 @@ public extension UIColor {
...
+        case .dark, .widgetDark:
+            return .gray650
+        case .black:
+            return .gray675
+        }
+    }
+
+    @objc public var overlayBackground: UIColor {
+        switch identifier {
+        case .light:
```
Comment (tonisevener) [inline WMF Framework/Theme.swift:201]: Maybe add `.widgetLight` to this case as well.

**Review by tonisevener — COMMENTED** [review]
Just a couple of minor things from looking at the old style widget color code (`WMFTodayContinueReadingWidgetViewController`), which I think is the only area that still uses `widgetLight` and `widgetDark` themes.

**Comment by tonisevener** [comment]
@mazevedofs I resolved all the "may be off" comments since I think a design sign off covers those. Found a couple of weird colors with the `hintWarningBackground`, I'll send them to you (for some reason GitHub is not allowing me to upload screenshots).

**Pull request merged**